receiver/prometheusreceiver/metrics_reciever_external_labels_test.go

```diff
@@ -62,46 +60,47 @@ func TestExternalLabels(t *testing.T) {
 	mp.wg.Wait()
 	metrics := cms.AllMetrics()
 
-	results := make(map[string][]*agentmetricspb.ExportMetricsServiceRequest)
+	// split and store results by target name
+	pResults := make(map[string][]*pdata.ResourceMetrics)
 	for _, md := range metrics {
 		rms := md.ResourceMetrics()
 		for i := 0; i < rms.Len(); i++ {
-			ocmd := &agentmetricspb.ExportMetricsServiceRequest{}
-			ocmd.Node, ocmd.Resource, ocmd.Metrics = opencensus.ResourceMetricsToOC(rms.At(i))
-			result, ok := results[ocmd.Node.ServiceInfo.Name]
+			name, _ := rms.At(i).Resource().Attributes().Get("service.name")
+			pResult, ok := pResults[name.AsString()]
 			if !ok {
-				result = make([]*agentmetricspb.ExportMetricsServiceRequest, 0)
+				pResult = make([]*pdata.ResourceMetrics, 0)
 			}
-			results[ocmd.Node.ServiceInfo.Name] = append(result, ocmd)
+			rm := rms.At(i)
+			pResults[name.AsString()] = append(pResult, &rm)
 		}
-
 	}
+	lres, lep := len(pResults), len(mp.endpoints)
+	assert.Equalf(t, lep, lres, "want %d targets, but got %v\n", lep, lres)
+
+	// loop to validate outputs for each targets
 	for _, target := range targets {
-		target.validateFunc(t, target, results[target.name])
+		validScrapes := getValidScrapes(t, pResults[target.name])
+		target.validateFunc(t, target, validScrapes)
 	}
 }
 
-func verifyExternalLabels(t *testing.T, td *testData, mds []*agentmetricspb.ExportMetricsServiceRequest) {
-	verifyNumScrapeResults(t, td, mds)
-
-	want := &agentmetricspb.ExportMetricsServiceRequest{
-		Node:     td.node,
-		Resource: td.resource,
+func verifyExternalLabels(t *testing.T, td *testData, rms []*pdata.ResourceMetrics) {
+	verifyNumScrapeResults(t, td, rms)
+	if len(rms) < 1 {
+		t.Fatal("At least one metric request should be present")
 	}
-	doCompare("scrape-externalLabels", t, want, mds[0], []testExpectation{
+	wantAttributes := td.attributes
+	metrics1 := rms[0].InstrumentationLibraryMetrics().At(0).Metrics()
+	ts1 := metrics1.At(0).Gauge().DataPoints().At(0).Timestamp()
+	doCompare("scrape-externalLabels", t, wantAttributes, rms[0], []testExpectation{
```

Comment:
Maybe not in this PR, I would like to see "t" as first argument.

Reply:
thanks @bogdandrutu I have one more PR coming right after this one for issue #6000 . I will update this in that PR.

Reply:
@bogdandrutu I have updated this comment in this PR, because It will be easier for upcoming Prometheus Receiver test cases to refer doCompare.